### PR TITLE
Master Dynamic I2C Allocation

### LINKED ISSUE #87: I2C Adres allocatie

### PR DESCRIPTION
**Pull request by NathanThus**
### Changes / Veranderingen
Implemented I2C dynamic adress allocation.

### Checklist

- [x] The code is linted / De code is gelint
- [x] The code is tested / De code is getest

### Issue

**Issue:** closes #87

### CODE REVIEW

File: Source/Master/src/main.cpp [hunk 1]
```diff
@@ -13,62 +13,86 @@
...
-    //   {
-    //     // Decompile with JSON parser
-    //     // Do some stuff with the data
+    #ifdef WIFI
+    if (webSocketHandler.isConnected())
+    {
+        std::string websocketData = webSocketHandler.Recieve();
+        if (websocketData != "")
+        {
+        // Decompile with JSON parser
```
Comment (TomVer99) [inline Source/Master/src/main.cpp:68]: I assume these will come later? or is this logic elsewhere?

File: Source/Master/src/main.cpp [hunk 2]
```diff
@@ -13,62 +13,86 @@
...
+        webSocketHandler.Send("Hello from ESP32!");
+    }
+    #endif
 
-    Node_Registers lastRegister = i2c.GetLastChange();
-    char data[MAX_STRING_SIZE];
+    #ifdef I2C_
+    if (i2c.IsAvailable())
+    {
+        if (i2c.GetLastChange() == LAST_ITEM_REGISTER) // TODO: Clean up
```
Comment (TomVer99) [inline Source/Master/src/main.cpp:81]: TODO?

**Review by TomVer99 — CHANGES_REQUESTED** [review]

File: Source/Master/lib/Communication Abstraction/IMasterSlave.hpp [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-{
-    NR_None = - 0x1,
-    NR_RequestForm = 0x0,
-    NR_Notification = 0x1, // | SOUND | VIBRATION | LED | RGB | 0 | 0 | 0 | 0 |
-    NR_RGB = 0x2, // | R | G | B | 0 | 0 | 0 | 0 | 0 |
-    NR_Item = 0x3, // Points to a char[] of size 48.
-    NR_Active = 0x4, // 0 = inactive, 1 = active
-    NR_Error = 0x5,
-};
+// enum Node_Registers
```
[the remark this answers is not in the document]
Comment (NathanThus) [reply Source/Master/lib/Communication Abstraction/IMasterSlave.hpp:4]: The conversion to a variable caused major issues within the recieveEvent switchcase.

[the remark this answers is not in the document]
Comment (NathanThus) [reply Source/Master/lib/Communication Abstraction/IMasterSlave.hpp:4]: Might be an idea to remove it entirely, however.

File: Source/Master/lib/I2C/I2C.cpp [hunk 4]
```diff
@@ -130,9 +138,128 @@ int I2C::GetRegister(Node_Registers my_register, void* data)
...
+
+    while (Wire.available() && i < MAX_STRING_SIZE)
+    {
+        data[i] = (char)Wire.read();
+        i++;
+    }
+
+    data[i] = '\0';
+
+    Serial.println("=== DATA ===");
```
[the remark this answers is not in the document]
Comment (NathanThus) [reply Source/Master/lib/I2C/I2C.cpp:207]: There are still a few prints, and can definetly prove useful with the demo, as another reference point.

File: Source/Master/src/main.cpp (hunk 1 above)
Comment (NathanThus) [reply Source/Master/src/main.cpp:68]: This is for another branch.

File: Source/Master/src/main.cpp (hunk 2 above)
Comment (NathanThus) [reply Source/Master/src/main.cpp:81]: Well, that has been cleaned up. I'll remove that 😉

File: Source/Master/lib/WiFi/WiFiHandler.cpp [hunk 5]
```diff
@@ -32,11 +32,11 @@ void WiFiHandler::Connect()
 
     if (WiFi.isConnected())
     {
-        Serial.println("Connected to WiFi");
+        // Serial.println("Connected to WiFi");
```
[the remark this answers is not in the document]
Comment (NathanThus) [reply Source/Master/lib/WiFi/WiFiHandler.cpp:35]: My bad!

File: Source/Master/lib/I2C/I2C.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (TomVer99) [reply Source/Master/lib/I2C/I2C.cpp:207]: Maybe add an option to disable / enable it? It looks like debug info?

[the remark this answers is not in the document]
Comment (NathanThus) [reply Source/Master/lib/I2C/I2C.cpp:207]: Fair enough, I'll surround it with #ifdef

**Comment by NathanThus** [comment]
*SHOULD* be all the changes implemented.

**Review by TomVer99 — APPROVED** [review]

**Pull request merged**